plugins/generator-1.15.2/forge-1.15.2/templates/fluid.java.ftl

```diff
@@ -209,17 +223,48 @@ import net.minecraft.block.material.Material;
 		</#if>
 	}
 
-	<#if data.spawnParticles>
+	<#if data.extendsForgeFlowingFluid()>
 	public static abstract class CustomFlowingFluid extends ForgeFlowingFluid {
 		public CustomFlowingFluid(Properties properties) {
 			super(properties);
 		}
 
+		<#if data.spawnParticles>
 		@OnlyIn(Dist.CLIENT)
 		@Override
 		public IParticleData getDripParticleData() {
 			return ${data.dripParticle};
 		}
+		</#if>
+
+		<#if data.flowStrength != 1>
+		@Override public Vec3d getFlow(IBlockReader world, BlockPos pos, IFluidState fluidstate) {
+			return super.getFlow(world, pos, fluidstate).scale(${data.flowStrength});
+		}
+		</#if>
+
+		<#if hasProcedure(data.canFlow)>
+		@Override protected boolean canFlow(IBlockReader worldIn, BlockPos fromPos, BlockState blockstate, Direction direction, BlockPos toPos, BlockState intostate, IFluidState toFluidState, Fluid fluidIn) {
+			boolean condition = true;
+			if (worldIn instanceof IWorld) {
+				int x = fromPos.getX();
+				int y = fromPos.getY();
+				int z = fromPos.getZ();
+				IWorld world = (IWorld) worldIn;
+				condition = <@procedureOBJToConditionCode data.canFlow/>;
+			}
+			return super.canFlow(worldIn, fromPos, blockstate, direction, toPos, intostate, toFluidState, fluidIn) && condition;

```

Comment:
I think that if custom condition is user, super call should be omitted, as this setup will add new conditions, not override existing flow conditions. Or is there some reason you set it like this?

Reply:
The super method does some checks (does the block have holes) that can't be done with procedures. Should I rename this condition to something like "Additional flowing condition"?

Reply:
Yes, that would be good to clear up the purpose of this condition better